include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -0,0 +1,318 @@
+                "compute_schur_complement",
+                Kokkos::MDRangePolicy<Kokkos::DefaultHostExecutionSpace, Kokkos::Rank<2>>(
+                        {0, 0},
+                        {m_bottom_right_block->size(), m_bottom_right_block->size()}),
+                [&](const int i, const int j) {
+                    double val = 0.0;
+                    for (int l = 0; l < m_top_left_block->size(); ++l) {
+                        val += m_bottom_left_block(i, l) * m_top_right_block(l, j);
+                    }
+                    m_bottom_right_block
+                            ->set_element(i, j, m_bottom_right_block->get_element(i, j) - val);
+                });
+    }
+
+public:
+    /**
+     * @brief Perform a pre-process operation on the solver. Must be called after filling the matrix.
+     *
+     * Block-LU factorize the matrix A according to the Schur complement method. The block-LU factorization is:
+     *
+     * A = |   Q    |             0             |  | I | Q^-1*gamma |
+     *     | lambda | delta - lambda*Q^-1*gamma |  | 0 |      I     |
+     *
+     * So we perform the factorization inplace to store only the relevant blocks in the matrix (while factorizing
+     * the blocks themselves if necessary):
+     *
+     * |   Q    |         Q^-1*gamma        |
+     * | lambda | delta - lambda*Q^-1*gamma |
+     */
+    void setup_solver() override
+    {
+        m_top_left_block->setup_solver();
+
+        // Compute Q^-1*gamma
+        auto top_right_block_device = create_mirror_view_and_copy(ExecSpace(), m_top_right_block);
+        m_top_left_block->solve(top_right_block_device);
+        deep_copy(m_top_right_block, top_right_block_device);
+
+        // Compute delta - lambda*Q^-1*gamma
+        compute_schur_complement();
+
+        m_bottom_right_block->setup_solver();
+    }
+
+    /**
+     * @brief Compute v <- v - lambda*u.
+     *
+     * [SHOULD BE PRIVATE (GPU programming limitation)]
+     *
+     * @param u
+     * @param v
+     */
+    virtual void solve_bottom_left_block_section(MultiRHS const u, MultiRHS v) const
+    {
+        auto bottom_left_block_device
+                = create_mirror_view_and_copy(ExecSpace(), m_bottom_left_block);
+        Kokkos::parallel_for(
+                "solve_bottom_left_block_section",
+                Kokkos::TeamPolicy<ExecSpace>(v.extent(1), Kokkos::AUTO),
+                KOKKOS_LAMBDA(
+                        const typename Kokkos::TeamPolicy<ExecSpace>::member_type& teamMember) {
+                    const int j = teamMember.league_rank();
+
+
+                    Kokkos::parallel_for(
+                            Kokkos::TeamThreadRange(teamMember, v.extent(0)),
+                            [&](const int i) {
+                                for (int l = 0; l < u.extent(0); ++l) {
+                                    v(i, j) -= bottom_left_block_device(i, l) * u(l, j);
+                                }
+                            });
```

Comment:
I guess it could take the form a parallel_reduce with a 2D TeamThreadMDRange.

Reply:
Oh yes that's better

Reply:
Actually a parallel_reduce on 2D array cannot compute `sum_l bottom_left_block_device(i, l) * u(l, j)`, it would compute `sum_{i,l} bottom_left_block_device(i, l) * u(l, j);`.

I think this discussion is a bit complicated so I switch on slack.